src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -243,6 +248,60 @@ void *frontendservicesesion(std::string masterIP, int connFd, SQLiteDBInterface
                 frontend_logger.log("Graph data file does not exist on the specified path", "error");
                 continue;
             }
+        } else if (line.compare(ADMDL) == 0) {
+            write(connFd, SEND.c_str(), FRONTEND_COMMAND_LENGTH);
+            write(connFd, "\r\n", 2);
+
+            char graph_data[FRONTEND_DATA_LENGTH];
+            bzero(graph_data, FRONTEND_DATA_LENGTH + 1);
+            string name = "";
+            string path = "";
+
+            read(connFd, graph_data, FRONTEND_DATA_LENGTH);
+
+            std::time_t time = chrono::system_clock::to_time_t(chrono::system_clock::now());
+            string uploadStartTime = ctime(&time);
+            string gData(graph_data);
+
+            Utils utils;
+            gData = utils.trim_copy(gData, " \f\n\r\t\v");
+            frontend_logger.log("Data received: " + gData, "info");
+
+            std::vector<std::string> strArr = Utils::split(gData, '|');
+
+            if (strArr.size() < 2) {
+                frontend_logger.log("Message format not recognized", "error");
+                continue;
+            }
+
+            name = strArr[0];
+            path = strArr[1];
+
+
+            if (JasmineGraphFrontEnd::modelExists(path, sqlite)) {
+                frontend_logger.log("Model exists", "error");
+                continue;
+            }
+
+            if (utils.fileExists(path)) {
+                frontend_logger.log("Path exists", "info");
+                std::string toDir = utils.getJasmineGraphProperty("org.jasminegraph.server.modelDir");
+                utils.copyToDirectory(path, toDir);
+
+                string sqlStatement =
+                        "INSERT INTO model (name,upload_path,upload_time,model_status_idmodel_status"
+                        ")VALUES(\"" + name + "\", \"" + path +
+                        "\", \"" + uploadStartTime + "\",\"" + to_string(Conts::GRAPH_STATUS::LOADING) + "\")";
+
+                int newModelID = sqlite.runInsert(sqlStatement);
+
+                frontend_logger.log("Upload done", "info");
+                write(connFd, DONE.c_str(), DONE.size());
+                write(connFd, "\n", 2);
+            } else {
+                frontend_logger.log("Graph data file does not exist on the specified path", "error");
```

Comment:
This error message should be on uploading the ML model and not on graph data file isn't it?

Reply:
Fixed in 2d8c122

Reply:
Done